lib/fog/vsphere/models/compute/server.rb

```diff
@@ -319,6 +320,13 @@ def initialize_scsi_controller
             Fog::Compute::Vsphere::SCSIController.new(self.attributes[:scsi_controller])
           end
         end
+
+        def initialize_virtual_machine_summary
+          if attributes[:summary] && attributes[:summary].props
+            self.attributes[:summary] = Fog::Compute::Vsphere::VirtualMachineSummary.new(self.attributes[:summary].props)
```

Comment:
I tried playing with this last week but was unable to get it going.  Other than that test, is there an example in another fog module I could look at to get it working?  I scanned through aws and google but didn't see it being used.